managedservices/managedservices.go

```diff
@@ -0,0 +1,230 @@
+package managedservices
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/giantswarm/apprclient"
+	"github.com/giantswarm/e2e-harness/pkg/framework"
+	frameworkresource "github.com/giantswarm/e2e-harness/pkg/framework/resource"
+	"github.com/giantswarm/helmclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Config struct {
+	Namespace string
+
+	ApprClient    *apprclient.Client
+	HelmClient    *helmclient.Client
+	HostFramework *framework.Host
+	Logger        micrologger.Logger
+}
+
+type ManagedServices struct {
+	namespace string
+
+	apprClient    *apprclient.Client
+	helmClient    *helmclient.Client
+	hostFramework *framework.Host
+	logger        micrologger.Logger
+	resource      *frameworkresource.Resource
+}
+
+func New(config Config) (*ManagedServices, error) {
+	var err error
+
+	if config.Namespace == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Namespace must not be empty", config)
+	}
+
+	if config.ApprClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ApprClient must not be empty", config)
+	}
+	if config.HostFramework == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.HostFramework must not be empty", config)
+	}
+	if config.HelmClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.HelmClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	var resource *frameworkresource.Resource
+	{
+		c := frameworkresource.ResourceConfig{
+			ApprClient: config.ApprClient,
+			HelmClient: config.HelmClient,
+			Logger:     config.Logger,
+			Namespace:  config.Namespace,
+		}
+
+		resource, err = frameworkresource.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	ms := &ManagedServices{
+		apprClient:    config.ApprClient,
+		helmClient:    config.HelmClient,
+		hostFramework: config.HostFramework,
+		logger:        config.Logger,
+		resource:      resource,
+	}
+
+	return ms, nil
+}
+
+func (ms *ManagedServices) Test(ctx context.Context, chartConfig ChartConfig, chartResources ChartResources) error {
+	var err error
+
+	err = validateChartConfig(chartConfig)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	err = validateChartResources(chartResources)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	{
+		ms.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("installing chart %#q", chartConfig.ChartName))
+
+		err = ms.resource.InstallResource(chartConfig.ChartName, chartConfig.ChartValues, chartConfig.ChannelName)
```

Comment:
Just saying. Seeing this it should rather read like `ms.resource.Install`.

Reply:
That's a good shout. We want to do refactoring in e2e-harness. Renaming this then makes sense to me.